pkg/referrerstore/oras/authprovider/authprovider.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authprovider
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/containerd/containerd/reference"
```

Comment:
Do we need to bring in containerd dependency?

Reply:
The only use for containerd is to leverage the reference parsing methods to extract the host name from the artifact string. Should we remove that and use an alternate way to extract the hostname?

Reply:
containerd is a heavy weight dependency and I would try to see if it can be avoided